### PR TITLE
Migrate to go modules

### LINKED ISSUE #72: Update to go modules
## Context

`dep` is officially being deprecated , the repo is already archived https://github.com/golang/dep 

I have already tried the initial update but ran into several issues.  It seems like we need to wait for this https://github.com/containerd/containerd/issues/3031 to be resolved, which is currently blocked on this PR https://github.com/containerd/cri/pull/1377#issuecomment-664136849 So progress is being made upstream.

### PR DESCRIPTION
**Pull request by LucasRoesler**
## Description

- Initial migration to go modules using the default output from i
  `go mod tidy`

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Resolves #72 

## How Has This Been Tested?

Deployed locally on swarm, deployed a function, and invoked it 

image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by LucasRoesler** [comment]
@alexellis once we merge this I will do the migration to buildx and github actions. That way i can replicate what we did in faas-netes. It will be easier to make it the same once the modules migration is done

**Review by alexellis — APPROVED** [review]
Approved

**Comment by alexellis** [comment]
I'm merge now to unblock you. Thanks for the PR

**Pull request merged**
